include/deal.II/base/symmetric_tensor.h

```diff
@@ -2039,6 +2064,49 @@ SymmetricTensor<rank_,dim,Number>::operator [] (const TableIndices<rank_> &indic
 
 
 
+template <int rank_, int dim, typename Number>
+inline
+Number *
+SymmetricTensor<rank_,dim,Number>::begin_raw()
+{
+  return &this->access_raw_entry(0);
+}
+
+
+
+template <int rank_, int dim, typename Number>
+inline
+const Number *
+SymmetricTensor<rank_,dim,Number>::begin_raw() const
+{
+  return &this->access_raw_entry(0);
+}
+
+
+
+template <int rank_, int dim, typename Number>
+inline
+Number *
+SymmetricTensor<rank_,dim,Number>::end_raw()
+{
+  Number *last_element_ptr = &(this->access_raw_entry(n_independent_components-1));
+  Number *past_last_element_ptr = ++last_element_ptr;
+  return past_last_element_ptr;
```

Comment:
Wouldn't it be simpler to do `return begin_raw()+n_independent_components`? You're proposing that this is a contiguous range anyway, so this may be easier and I think it would also help the compiler propagate some constants.